src/navigation/test/url-serializer.spec.ts

```diff
@@ -797,6 +799,21 @@ describe('UrlSerializer', () => {
       expect(segmentPairs[2].segments[1].name).toEqual('taco-page');
       expect(segmentPairs[2].segments[1].secondaryId).toEqual('fifth-page');
     });
```

Comment:
I like that a failing test was added to show the error, but I don't think we should merge a PR that just has a failing test w/o the code that then fixes the failing test.

Reply:
That said, I am not sure that I agree with the validity of the test and issue.

If we have two pages, OneThing and TwoThing defined as such:

```ts
@IonicPage({
  name: 'one-thing-page',
  segment: 'thing',
  defaultHistory: ['foo']
})
...
export class OneThing {}
```

```ts
@IonicPage({
  name: 'two-thing-page',
  segment: 'thing',
  defaultHistory: ['bar']
})
...
export class TwoThing {}
```

From within `bar`, We could do a `this.navCtrl.push('OneThing');` or a `this.navCtrl.push('TwoThing');` resulting in an url of `/#/bar/thing` but with two different "thing pages" which does not seem like a good idea in the first place.

Given that the code here is parsing an URL and determining what page to show, I can buy the argument that we should take the default history into account, but it doesn't seem right that we should be allowing the same URL to potentially represent different pages to begin with.

Not saying anything here is right or wrong, just thinking out-loud that we may have a deeper issue here.

Reply:
@kensodemann My use case for `defaultHistory` is to try to achieve a specific page hierarchy with clean urls.  While the developer could push `one-thing-page` onto `bar`, it would violate the hierarchy the developer is trying to achieve.

My use of `IonicPage` is simply for routing purposes.  The naming convention of the pages follows the hierarchy of what I am I trying to achieve.  In your specific example, I would rename the pages to `foo.thing` and `bar.thing`.  My root pages would be named `foo` and `bar`.  Therefore if I am within the root of `bar`, it wouldn't make sense to push `foo.thing` as you have explained in your example.

Having the same segment for different pages is a must have requirement.  In my example above, I may wish to have a `settings` page for `foo` and `bar`.  I would like my urls to look like `/#/foo/settings` and `/#/bar/settings`, however, this is not currently possible.